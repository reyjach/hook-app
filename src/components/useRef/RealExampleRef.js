import React, { useState } from 'react';
import { MultiplesCutomerHook } from '../examples/MultiplesCutomerHook';
import '../useEffect/effect.css';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    const handleClick = () => {


        setShow(!show);
    }

  return (
    <div>
        <h1> Real Example Ref</h1>
        <hr />

        { show && <MultiplesCutomerHook />}

        <button 
            className="btn btn-outline-primary mt-3"
            onClick={handleClick}
        >
                Show hide
        </button>
    </div>
  )
}
