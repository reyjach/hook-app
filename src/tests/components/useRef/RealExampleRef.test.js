import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/useRef/RealExampleRef";

describe('Pruebas en el componente RealExampleRef', () => { 

    const wrapper = shallow(<RealExampleRef />);

    test('Debe mostrarse correctamente', () => { 

        

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultiplesCutomerHook').exists() ).toBe(false);

     });

     test('Debe de mostrar el componente MultiplesCutomerHook', () => { 

        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultiplesCutomerHook').exists() ).toBe(true);


     })
 })