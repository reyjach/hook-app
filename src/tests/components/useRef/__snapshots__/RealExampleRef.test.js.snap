// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
     Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    Show hide
  </button>
</div>
`;
